core/src/autogluon/core/_setup_utils.py

```diff
@@ -18,7 +18,7 @@
 DEPENDENT_PACKAGES = {
     "boto3": ">=1.10,<2",  # <2 because unlikely to introduce breaking changes in minor releases. >=1.10 because 1.10 is 3 years old, no need to support older
     "numpy": ">=1.21,<1.29",  # "<{N+3}" upper cap, where N is the latest released minor version, assuming no warnings using N
-    "pandas": ">=2.0.0,<2.2.0",  # "<{N+1}" upper cap
+    "pandas": ">=2.0.0,<2.3.0",  # "<{N+1}" upper cap
```

Comment:
Pandas 2.2 introduced a breaking change to how time frequency aliases are represented as strings (https://pandas.pydata.org/docs/whatsnew/v2.2.0.html#other-deprecations). For example, `pandas.tseries.frequencies.to_offset("1min")` returns 
- `min` in pandas >=2.2
- `T` in pandas <2.2
I will need to change our logic for detecting seasonality to take this into account.

Reply:
Moving pandas upgrade to a separate PR to isolate the change